thread.c

```diff
@@ -3732,6 +3729,11 @@ rb_thread_scheduler_set(VALUE thread, VALUE scheduler)
 
     VM_ASSERT(th);
 
+    // We invoke Scheduler#close when setting it to something else, so that any state relating to outstanding locks is cleaned up.
```

Comment:
I'd suggest:
>  We invoke Scheduler#close when setting it to something else, to ensure the previous scheduler runs to completion before changing the scheduler. That way, we do not need to consider interactions, e.g., of a Fiber from the previous scheduler with the new scheduler.

Reply:
I am going to do a documentation update so I'll apply this.